ionoscloud/resource_dbaas_mariadb_cluster.go

```diff
@@ -171,7 +179,15 @@ func mariaDBClusterCreate(ctx context.Context, d *schema.ResourceData, meta inte
 func mariaDBClusterDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	client := meta.(services.SdkBundle).MariaDBClient
 	clusterID := d.Id()
-	_, apiResponse, err := client.DeleteCluster(ctx, d.Id())
+	var location string
+	// This takes care of the old version of the provider where the location was not required and
+	// all the requests were done for "de/txl" location with the correspondent endpoint.
+	if locationValue, locationOk := d.GetOk("location"); locationOk {
```

Comment:
doesn't `location` being Required mean `GetOk` is redundant here ?

Reply:
@cbeti-ionos no, this takes care of the case in which the user has an older version of the provider that doesn't have the `location` attribute.